File: app-presentation/src/main/java/com/zxjdev/smile/presentation/communal/util/image/GlideImageLoader.java
package com.zxjdev.smile.presentation.communal.util.image;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.DrawableRes;
import android.support.v4.app.Fragment;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestManager;

import jp.wasabeef.glide.transformations.CropCircleTransformation;

public class GlideImageLoader implements ImageLoader {

  private Context context;
  private RequestManager requestManager;

  public GlideImageLoader(Context context, RequestManager requestManager) {
    this.context = context;
    this.requestManager = requestManager;
  }

  public GlideImageLoader(Activity activity) {
    this.context = activity.getApplicationContext();
    this.requestManager = Glide.with(activity);
  }

  public GlideImageLoader(Fragment fragment) {
    this.context = fragment.getActivity().getApplicationContext();
    this.requestManager = Glide.with(fragment);
  }

  @Override
  public void loadImage(String url, ImageView imageView) {
    requestManager.load(url).into(imageView);
  }

  @Override
  public void loadImage(@DrawableRes int res, ImageView imageView) {
    requestManager.load(res).into(imageView);
  }

  @Override
  public void loadCircleImage(String url, ImageView imageView) {
    requestManager.load(url).bitmapTransform(new CropCircleTransformation(context)).into(imageView);
  }

  @Override
  public void loadCircleImage(@DrawableRes int res, ImageView imageView) {
    requestManager.load(res).bitmapTransform(new CropCircleTransformation(context)).into(imageView);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/util/image/GlideImageLoader.kt
package com.zxjdev.smile.presentation.common.util.image

import android.app.Activity
import android.content.Context
import androidx.annotation.DrawableRes
import androidx.fragment.app.Fragment
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.bumptech.glide.RequestManager

import jp.wasabeef.glide.transformations.CropCircleTransformation

class GlideImageLoader : ImageLoader {

    private var context: Context? = null
    private var requestManager: RequestManager? = null

    constructor(context: Context, requestManager: RequestManager) {
        this.context = context
        this.requestManager = requestManager
    }

    constructor(activity: Activity) {
        this.context = activity.applicationContext
        this.requestManager = Glide.with(activity)
    }

    constructor(fragment: Fragment) {
        this.context = fragment.activity?.applicationContext
        this.requestManager = Glide.with(fragment)
    }

    override fun loadImage(url: String, imageView: ImageView) {
        requestManager!!.load(url).into(imageView)
    }

    override fun loadImage(@DrawableRes res: Int, imageView: ImageView) {
        requestManager!!.load(res).into(imageView)
    }

    override fun loadCircleImage(url: String?, imageView: ImageView) {
        requestManager!!.load(url).transform(CropCircleTransformation(context)).into(imageView)
    }

    override fun loadCircleImage(@DrawableRes res: Int, imageView: ImageView) {
        requestManager!!.load(res).transform(CropCircleTransformation(context)).into(imageView)
    }
}
